### ISSUE TITLE
Suite timeout in org.elasticsearch.discovery.ClusterDisruptionIT on Windows CI (might be  java.lang.AssertionError: global checkpoint [-1] for primary mode allocation ID)

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#26707]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+multijob-windows-compatibility/174/consoleFull

The log is fairly huge but this is the first thing I see that look incriminating:
```
 sep 18, 2017 11:45:56 PM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
 WARNING: Uncaught exception in thread: Thread[elasticsearch[node_t0][clusterApplierService#updateTask][T#1],5,TGRP-ClusterDisruptionIT]
 java.lang.AssertionError: global checkpoint [-1] for primary mode allocation ID [Cqh0BxHnRiuFhvO8JpdtHg] less than in-sync global checkpoints [{-TgCKmeaRDyW7gCNyPUAeQ=LocalCheckpointState{localCheckpoint=-2, globalCheckpoint=-2, inSync=true}, G22BRgsaSXa0EGpS4TzsZw=LocalCheckpointState{localCheckpoint=2, globalCheckpoint=1, inSync=true}, Cqh0BxHnRiuFhvO8JpdtHg=LocalCheckpointState{localCheckpoint=2, globalCheckpoint=-1, inSync=true}}]
 	at __randomizedtesting.SeedInfo.seed([B93BBCA37C85D2DE]:0)
 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.invariant(GlobalCheckpointTracker.java:284)
 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.updateFromMaster(GlobalCheckpointTracker.java:455)
 	at org.elasticsearch.index.seqno.SequenceNumbersService.updateAllocationIdsFromMaster(SequenceNumbersService.java:235)
 	at org.elasticsearch.index.shard.IndexShard.updateShardState(IndexShard.java:388)
 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.updateShard(IndicesClusterStateService.java:575)
 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.createOrUpdateShards(IndicesClusterStateService.java:522)
 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:222)
 	at org.elasticsearch.cluster.service.ClusterApplierService.lambda$callClusterStateAppliers$6(ClusterApplierService.java:495)
 	at java.lang.Iterable.forEach(Iterable.java:75)
 	at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:492)
 	at org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:479)
 	at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:429)
 	at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:158)
 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569)
 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247)
 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
 	at java.lang.Thread.run(Thread.java:745)
 And I lost the rest of the exception trace but I don't think that is much left.
```

OTOH, the test intentionally does all kinds of incriminating stuff so I dunno if this is intentional.

**Comment by nik9000** [comment]
On second thought, given the location of the assertion I doubt this is intentional. We don't tend to catch assertions in production code in the tests.

**Comment by nik9000** [comment]
Another: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+multijob-unix-compatibility/os=centos/174/console

**Comment by nik9000** [comment]
Another:  https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+periodic/38/consoleFull

```
02:23:44   2> sept. 19, 2017 7:34:25 AM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
02:23:44   2> WARNING: Uncaught exception in thread: Thread[elasticsearch[node_t0][__mock_network_thread][T#5],5,TGRP-ClusterDisruptionIT]
02:23:44   2> java.lang.AssertionError: global checkpoint [7] for primary mode allocation ID [ZsYjlrZBQcmXpqfqKbf2CQ] less than in-sync global checkpoints [{ZsYjlrZBQcmXpqfqKbf2CQ=LocalCheckpointState{localCheckpoint=12, globalCheckpoint=7, inSync=true}, oJWVx9F7SmOxWkU-kASlLA=LocalCheckpointState{localCheckpoint=-2, globalCheckpoint=-2, inSync=true}, MA_qGeBQRZCODqgYq0QR8g=LocalCheckpointState{localCheckpoint=8, globalCheckpoint=8, inSync=true}}]
02:23:44   2> 	at __randomizedtesting.SeedInfo.seed([D9EF70DDE745DDB6]:0)
02:23:44   2> 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.invariant(GlobalCheckpointTracker.java:284)
02:23:44   2> 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.updateGlobalCheckpointForShard(GlobalCheckpointTracker.java:417)
```

**Comment by nik9000** [comment]
Another: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+aggressive-opts/233/consoleFull

**Comment by nik9000** [comment]
More: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/351/consoleFull

```
15:04:36   2> sep 19, 2017 11:44:32 PM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
15:04:36   2> WARNING: Uncaught exception in thread: Thread[elasticsearch[node_t2][es_nio_transport_worker][T#1],5,TGRP-ClusterDisruptionIT]
15:04:36   1> --------[test][0], node[null], [R], recovery_source[peer recovery], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2017-09-19T14:44:32.774Z], delayed=true, details[node_left[hyasxLnqRaGaxjDh1VE1zQ]], allocation_status[no_attempt]]
15:04:36   2> java.lang.AssertionError: global checkpoint [1] for primary mode allocation ID [4mXtmtFPSBSdcKriF9Peng] less than in-sync global checkpoints [{3SiSPLmRQiySdQt7jG3C8Q=LocalCheckpointState{localCheckpoint=-2, globalCheckpoint=-2, inSync=true}, 4mXtmtFPSBSdcKriF9Peng=LocalCheckpointState{localCheckpoint=4, globalCheckpoint=1, inSync=true}, UlauiSosRfC70YNJwQw0Iw=LocalCheckpointState{localCheckpoint=4, globalCheckpoint=3, inSync=true}}]
15:04:36   2> 	at __randomizedtesting.SeedInfo.seed([14B6B875C08506C2]:0)
15:04:36   2> 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.invariant(GlobalCheckpointTracker.java:284)
15:04:36   2> 	at org.elasticsearch.index.seqno.GlobalCheckpointTracker.updateGlobalCheckpointForShard(GlobalCheckpointTracker.java:417)
15:04:36   2> 	at org.elasticsearch.index.seqno.SequenceNumbersService.updateGlobalCheckpointForShard(SequenceNumbersService.java:138)
15:04:36   1> customs:
15:04:36   1>    snapshot_deletions: SnapshotDeletionsInProgress[]   snapshots: SnapshotsInProgress[]   restore: RestoreInProgress[])
15:04:36   2> 	at org.elasticsearch.index.shard.IndexShard.updateGlobalCheckpointForShard(IndexShard.java:1686)
15:04:36   1> [2017-09-19T23:44:32,781][TRACE][o.e.a.b.TransportShardBulkAction] [node_t2] send action [indices:dat
```

**Comment by jasontedor** [comment]
Closed by 581a873124422bcbbb3cb630ca971df22550593c
